mne_bids/tests/test_path.py

```diff
@@ -560,10 +531,10 @@ def test_bids_path(return_bids_test_dir):
     assert bids_path.basename == 'sub-01_ses-02_task-03_split-01_ieeg.mat'
 
     # test home dir expansion
-    bids_path = BIDSPath(root='~/foo')
+    bids_path = BIDSPath(root='~/foo', subject='test')
```

Comment:
Why do we need this?

Reply:
Because of these lines since now `root='.'` by default:

```        if all(ii is None for ii in [subject, session, task,
                                     acquisition, run, processing,
                                     recording, space, root, suffix,
                                     recording, space, suffix, root,
                                     extension]):
            raise ValueError("At least one parameter must be given.")
```

Reply:
Yes I've seen that, but why do you set subject now?

Reply:
Oh ok I see. Mhh. Not sure if I like that... need to think 🤔

Reply:
Yeah agreed. It's a weird situation. :( I will try a few other things and ping you then... 

Do you have any thoughts on how to address the `emptyroom` stuff? Downstream, `emptyroom` stuff is using `BIDSPath.fpath` with the inferring suffix/extension functionality, which will break now.

Reply:
Let's discuss this tomorrow, I'm still drying my computer.... :S

Reply:
Just saw that... :(

Idk if helpful, but when I drop my phone in water, it helps to cover it with rice for a night (soaks out all the water). Possibly helpful for laptop?

Reply:
Seems to be working alright again! thanks :)